### DISCUSSION TITLE
How to keep track of the openAI tokens used by a session?

### DISCUSSION THREAD

**Discussion by saikarthikp9** [FlowiseAI/Flowise#940] [Q&A]
As I understand, I can create a tool to make API calls, so I can make API calls to my custom backend to store token usage. But how can I get this info from the Conversational Retrieval QA Chain output?

**Comment by mattbisme** [comment]
I've been trying to figure out the same thing.

**Comment by amansoniamazatic** [comment]
@saikarthikp9 @mattbisme Use LangSmith or LLM Monitor API to keep track of all input and output token usage.

**Comment by chungyau97** [comment]
Hi all, we will need to make some enhancement for this to work.

Extracting the value of `ChatId` and overide the value of LangSmith/LangFuse `Project Name` to keep track on each ChatEmbed sessions.

I have added this to our backlog.
